enhancements/insights/on-demand-data-gathering.md

```diff
@@ -0,0 +1,255 @@
+4. The new container process update the status attributes in the corresponding CR instance and stops.
+
+This scenario brings few questions and things to be discussed. Please see the following sections [Implementation details](#implementation-detailsnotesconstraints), [Risk mitigations](#risks-and-mitigations) and [Open questions](#open-questions)
+
+### API Extensions
+
+This is adding new Insights CRD in the `insights.openshift.io/v1alpha1` . The proposal of the structure of the new CRD is mentioned in the [Proposal](#Proposal) section.
+
+### Implementation Details/Notes/Constraints
+
+The Insights operator currently runs and gathers the data in a single container. 
+Uploading of new archives, pruning old archives as well as downloading the last Insights analysis report (this is always per archive) is implemented in this one container/binary. 
+When there is a successful upload (to the console.redhat.com) of the archive then there is some short delay (waiting for the processing in console.redhat.com pipelines) and then there is request to get the last Insights analysis report per given cluster. 
+Finally the status section of the `insightsoperator.operator.openshift.io` CR is updated. Note that there is a retry logic in case of failed upload and after some time (and number of attempts) the Insights operator is marked as degraded.
+Introducing a new on demand process/container for running an "extra" gather process raises the question about parallelism. It should be feasible to allow two (or more) parallel gather processes, but it is very important to keep in mind that only one (and that is the latest) Insights analysis report will be available.  
+
+
+### Risks and Mitigations
+
+Risk: Inconsistent Insights analysis results. Assume following situation:
+- periodic gathering is running in the Insights operator (let's call it gather A)
+- user runs a new on demand data gathering (let's call it gather B) e.g 30 seconds after gather A
+- gather A and B finish at essentially the same time
+- the external processing service (in the console.redhat.com) provides only one recent Insights analysis result, and there is no guarantee which (of the two) it is.
+- this can be very confusing, because user running gather B can get result of the gather A (e.g there could be different gatherers enabled, which would produce different results)
+
+Mitigation: This is the very first question in the [Open questions](#open-questions) section, but perhaps reporting the Insights analysis result in one place could slightly mitigate this problem (the status part could tell whether it was a periodic gather or user-triggered gather). 
+Another option would be to disallow the parallel gather execution - if the Insights operator sees a new CR instance (or any other running gather process) then it interrupts any gathering in progress. 
```

Comment:
This seems sub-optimal to me, I think having a queue would be preferred here over cancelling and starting the newer one.

If the user wants to interrupt the gather event they can delete the `DataGather`, but a new one being created, I would expect, joins the queue

Reply:
OK. I described my opinion about the queue approach in the https://github.com/openshift/enhancements/pull/1279/files#r1022423877.  I'm not convinced it does any good.

Reply:
Maybe we could limit the size of the queue to 1, but it still seems to me more straightforward to interrupt any active gathering process (no extra upload, download of the data).

Reply:
OK so the queue it is.